File: GeneratedEntityReaderClasses/Components/PressurePlateComponent.cs
namespace GeneratedNoitaClasses;
public class PressurePlateComponent : NoitaComponentBase,  iNoitaType<PressurePlateComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt check_every_x_frames { get; set; } = new();
	public NoitaInt state { get; set; } = new();
	public NoitaCVector2<NoitaFloat> aabb_min { get; set; } = new();
	public NoitaCVector2<NoitaFloat> aabb_max { get; set; } = new();
	public NoitaFloat material_percent { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		check_every_x_frames.Read(s);
		state.Read(s);
		aabb_min.Read(s);
		aabb_max.Read(s);
		material_percent.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		check_every_x_frames.Write(s);
		state.Write(s);
		aabb_min.Write(s);
		aabb_max.Write(s);
		material_percent.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/PressurePlateComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaCVector2
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class PressurePlateComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    check_every_x_frames:NoitaInt = field(default_factory=NoitaInt)
    state:NoitaInt = field(default_factory=NoitaInt)
    aabb_min:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    aabb_max:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    material_percent:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.check_every_x_frames.Read(s)
        self.state.Read(s)
        self.aabb_min.Read(s)
        self.aabb_max.Read(s)
        self.material_percent.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.check_every_x_frames.Write(s)
        self.state.Write(s)
        self.aabb_min.Write(s)
        self.aabb_max.Write(s)
        self.material_percent.Write(s)

